lib/daru/category.rb

```diff
@@ -131,11 +131,17 @@ def add_category(*new_categories)
     #   dv = Daru::Vector.new [:a, 1, :a, 1, :c], type: :category
     #   dv.count :a
     #   # => 2
-    def count category
-      raise ArgumentError, "Invalid category #{category}" unless
-        categories.include?(category)
+    #   dv.count
+    #   # => 5
+    def count category = nil
+      if category.nil? # count all
+        @cat_hash.values.map(&:size).sum
+      else
+        raise ArgumentError, "Invalid category #{category}" unless
+          categories.include?(category)
 
-      @cat_hash[category].size
+        @cat_hash[category].size
+      end
```

Comment:
Three notes:

1. Let's rely on early return idiom:
```ruby
return ... if category.nil?
rais ... unless
@cat_hash[category].size
```
2. Let's not rely on `sum` method, it is not in stdlib before 2.4. It accidentally works (because of active_support required in specs, and ruby-ole required in lib), but it is pure accident.
3. Maybe some dedicated "undefined" default value would be more reliable? Can't we have categorical vector with `nil`s?